chronos/streams/asyncstream.nim

```diff
@@ -141,6 +141,24 @@ proc copyData*(sb: AsyncBuffer, dest: pointer, offset, length: int) {.inline.} =
   copyMem(cast[pointer](cast[uint](dest) + cast[uint](offset)),
           unsafeAddr sb.buffer[0], length)
 
+proc upload*(sb: ptr AsyncBuffer, pbytes: ptr byte,
```

Comment:
If there is a reason why `openArray` cannot be used here, let's add a comment about it. 
If there is no such reason, we should always prefer to use `openArray` as it has the potential to make the code provably memory-safe in the future (that is, in [future versions of Nim](https://forum.nim-lang.org/t/5289)).

Reply:
@zah this procedure is `{.async.}` and i wish to change it to something else, but currently its impossible.

Reply:
Ah, right, pity. Still, a TODO comment may be appropriate in such places, because we might be able to remove the openarray restriction in the future.